svgnative/include/svgnative/SVGRenderer.h

```diff
@@ -104,20 +104,113 @@ using Paint = boost::variant<Color, Gradient>;
 using ColorStop = std::pair<float, Color>;
 using ColorMap = std::map<std::string, Color>;
 
-struct Rect
+/**
+ * Represents a closed interval.
+ *
+ * TODO: Just remove these mathematical definitions?
+ *
+ * The following statements hold true for an interval:
+ * 1. An interval $I$ is defined as the set of all numbers between two
+ * numbers $u$ and $v$ on the number line, but not equal to either of them.
+ * 2. If $u$ and $v$ are the same numbers, the set becomes empty and thus
+ * the interval is also recognized as empty.
+ * 3. An interval $I$ is said to contain an interval $C$ if each element in
+ * the set of interval $C$ is also a member of the set of interval $I$.
+ * 4. The join of two intervals $A$ and $B$ is defined as an interval whose
+ * set is defined as the smallest possible set of *complete* real numbers that
+ * contain all members of $A$ as well as $B$. By complete we simply mean that
+ * for any possible two members of the set, all real numbers between the two
+ * must also be within the set. The word smallest here means that no other proper
+ * subset of that set will satisfy all the requirements.
+ * 5. The intersection of two intervals $A$ and $B$ is the one whose set is
+ * the intersection of the sets of $A$ and $B$.
+ */
+class Interval
```

Comment:
Didn't get to review how Interval is used yet, but it seems that it could be a private header and doesn't need to be in public headers.

Reply:
Ah ok, an Interval seems to be a coordinate pair.

Maybe, instead of `x`, `y`, `width`, `height` we should use `x1`, `y1` and `x2`, `y2` on `Rect` directly.

Ideally, all the logic would then be hidden and the changes to the Rect class are as minimal as possible.

Reply:
@dirkschulze Umm, so interval is basically a supporting class that makes life easier for rectangle intersections and joins. It doesn't need to be used by the client applications, `Rect` is public however and  it has a function that gets the intervals. This is used internally so maybe I can make it private.
> Maybe, instead of x, y, width, height we should use x1, y1 and x2, y2 on Rect directly.
Ideally, all the logic would then be hidden and the changes to the Rect class are as minimal as possible.

Could you elaborate this? If we simply replace width and height by x2 and y2, we have to compute width and height in function calls for those place where they are needed. That avoids computation for sure, but how does it avoid changes to the `Rect` class?